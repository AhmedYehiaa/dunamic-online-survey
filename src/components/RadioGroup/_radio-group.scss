.radio-group {
  min-width: 75%;
}