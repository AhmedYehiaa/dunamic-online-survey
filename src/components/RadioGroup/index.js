import React from 'react';
import PropTypes from 'prop-types';
import RadioOption from '../RadioOption';

const RadioGroup = ({
  name, selectedValue, options, onClick,
}) => (
  <div className="radio-group">
    {options.map(option => (
      <RadioOption
        key={option.value}
        name={name}
        label={option.name}
        value={option.value}
        onClick={onClick}
        defaultChecked={selectedValue === option.value}
      />
    ))}
  </div>
);

RadioGroup.propTypes = {
  name: PropTypes.string.isRequired,
  selectedValue: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
  })).isRequired,
  onClick: PropTypes.func.isRequired,
};

RadioGroup.defaultProps = {
  selectedValue: '',
};

export default RadioGroup;
