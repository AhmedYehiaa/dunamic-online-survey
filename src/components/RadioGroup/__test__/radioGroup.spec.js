import React from 'react';
import { shallow } from 'enzyme';
import RadioGroup from '../index';

it('it should renders the radio options', () => {
  const handleUpdatingAnswer = jest.fn();
  const value = 'value1';
  const options = [{
    name: 'name1',
    value: 'value1',
  }];
  const wrapper = shallow(
    <RadioGroup
      name="productSurvey"
      selectedValue={value}
      value={value}
      options={options}
      onClick={handleUpdatingAnswer}
    />,
  );
  expect(wrapper.children.length).toEqual(1);
  expect(wrapper).toMatchSnapshot();
});
