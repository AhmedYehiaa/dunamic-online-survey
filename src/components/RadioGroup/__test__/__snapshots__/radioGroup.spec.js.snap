// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should renders the radio options 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RadioGroup
    name="productSurvey"
    onClick={[MockFunction]}
    options={
      Array [
        Object {
          "name": "name1",
          "value": "value1",
        },
      ]
    }
    selectedValue="value1"
    value="value1"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RadioOption
          defaultChecked={true}
          label="name1"
          name="productSurvey"
          onClick={[MockFunction]}
          value="value1"
        />,
      ],
      "className": "radio-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "value1",
        "nodeType": "function",
        "props": Object {
          "defaultChecked": true,
          "label": "name1",
          "name": "productSurvey",
          "onClick": [MockFunction],
          "value": "value1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RadioOption
            defaultChecked={true}
            label="name1"
            name="productSurvey"
            onClick={[MockFunction]}
            value="value1"
          />,
        ],
        "className": "radio-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "value1",
          "nodeType": "function",
          "props": Object {
            "defaultChecked": true,
            "label": "name1",
            "name": "productSurvey",
            "onClick": [MockFunction],
            "value": "value1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
